### PR TITLE
Rework of CLI + Cobra

### LINKED ISSUE #56: New CLI output format
Some mockups for sbctl interface that fits better with existing *ctl tooling.

Inspiration:
    * boltctl
    * fwupdmgr
    * bootctl

<pre>
✔ (Green)       Positive/Succsess
✘ (Red)         Negative/Error
‼ (Yellow)      Warning
⁇ (Yellow(?))   Unknwon/Exception/Somethingweird
</pre>

# Errors/Exceptions
<pre>
λ bin » sbctl list-bundles
⁇ It might be necessary to run this tool as root:
⁇ open /usr/share/secureboot/bundles.db: permission denied
(Bold red?)
</pre>

# Status
<pre>
$ sbctl status
✔ Secure Boot: Disabled
✔ Setup Mode:  User

Curent boot entry
✔ Signed / ✘ Not Signed (!)
    Title:		Arch Linux (systemd-linux.efi)
    Id:			systemd-linux.efi
    Source:		/efi/EFI/Linux/systemd-linux.efi
    Linux:		EFI/Linux/systemd-linux.efi
    Options:	rw quiet acpi_osi= acpi_backlight=thinkpad_screen nowatchdog bgrt_disable lockdown=confidentiality

json = {
    "Secure Boot": true,
    "Setup Mode": "user"
}
</pre>

# Verify
<pre>
$ sbctl verify
Verifying file database and EFI images in /efi...
✔ signed /boot/vmlinuz-linux
✔ signed /efi/EFI/BOOT/BOOTX64.EFI 
✔ signed /efi/EFI/Linux/linux-linux.efi
✔ signed /efi/EFI/Linux/systemd-linux.efi 
✔ signed /efi/EFI/arch/fwupdx64.efi 
✔ signed /efi/EFI/systemd/systemd-bootx64.efi 
✔ signed /usr/lib/fwupd/efi/fwupdx64.efi.signed 
✘ could not sign /efi/EFI/Linux/linux-linux-mitigations.efi

[
    {
        "file": "/boot/vmlinuz-linux"
        "is_signed": True,
    }
]
</pre>

# List Files
<pre>
$ sbctl list-files
Enrolled files:

/efi/EFI/arch/fwupdx64.efi
    Signed:		✔ Signed / ✘ Not Signed (!)

/efi/EFI/systemd/systemd-bootx64.efi
    Signed:		✔ Signed / ✘ Not Signed (!)

/usr/lib/fwupd/efi/fwupdx64.efi
    Output:		/usr/lib/fwupd/efi/fwupdx64.efi.signed
    Signed:		✔ Signed / ✘ Not Signed (!)

/boot/vmlinuz-linux
    Signed:		✔ Signed / ✘ Not Signed (!)

/efi/EFI/BOOT/BOOTX64.EFI
    Signed:		✔ Signed / ✘ Not Signed (!)

/efi/EFI/Linux/linux-linux.efi
    Signed:		✔ Signed / ✘ Not Signed (!)

/efi/EFI/Linux/systemd-linux.efi
    Signed:		✔ Signed / ✘ Not Signed (!)
</pre>

# List Bundles
<pre>
$ sudo sbctl list-bundles
Enrolled bundles:

/efi/EFI/Linux/linux-linux.efi
    ESP Location:       /efi
    Output:             └─/EFI/Linux/linux-linux.efi
    EFI Stub Image:       └─/usr/lib/systemd/boot/efi/linuxx64.efi.stub
    Splash Image:           ├─/usr/share/systemd/bootctl/splash-arch.bmp
    Cmdline:                ├─/etc/kernel/cmdline
    OS Release:             ├─/usr/lib/os-release
    Kernel Image:           ├─/boot/vmlinuz-linux
    Initramfs Image:        └─/boot/initramfs-linux.img
    Intel Microcode:          └─/boot/intel-ucode.img
    Signed:             ✔ Signed / ✘ Not Signed (!)
</pre>

### LINKED ISSUE #59: Replace `log.Fatal` with `log.Panic`?
`log.Fatal`:
```
2021/05/04 15:10:55 exit status 1
exit status 1
```
VS

`log.Panic`:
```
2021/05/04 15:09:17 exit status 1
panic: exit status 1

goroutine 1 [running]:
log.Panic(0xc0000b3dc8, 0x1, 0x1)
	/usr/lib/go/src/log/log.go:354 +0xae
github.com/foxboron/sbctl.GetESP(0x656580, 0xc000066840)
	/home/user/Projects/sbctl/sbctl.go:26 +0x176
main.bundleCmd(0x7e1d00)
	/home/user/Projects/sbctl/cmd/sbctl/main.go:235 +0x249
main.main()
	/home/user/Projects/sbctl/cmd/sbctl/main.go:338 +0x1fc
exit status 2
```

I think having trackbacks is very nice for debugging and bug reports.

### LINKED ISSUE #62: Shell completions are corrupted.
Screenshot from 2021-05-05 10-09-33

All of shell completions have this binary string at the end.

In `less` it looks like this:

`ESC(BESC[m`

Probably an issue with upstream library?

### PR DESCRIPTION
**Pull request by Foxboron**
This is a massive and probably quite impossible to review properly. But lets try :)

This restructures most of the top-level cobra commands with the goal of having proper error propagation and seperation between user-facing interaction and keeping some sbctl functionality as a library.

All of the printing has been reworked. All errors are properly bubbled up, but some might not leave a user friendly error in `main.go` yet.  Missing is propper debug logging infrastructure, and key generation/enrollment is left as-is because we are moving to `go-uefi` anyway.

Fixes #59
Fixes #62
Fixes #56

### CODE REVIEW

**Comment by Foxboron** [comment]
This is also missing proper flag grouping. For the sake of reference the inspiration for the layout has been taken from podman :)

https://github.com/containers/podman/blob/master/cmd/podman/containers/rm.go

**Comment by Foxboron** [comment]
It's also missing `--json` being a bit more granular.  It's added to all commands and hides all prints by default, however not all commands have a `json` output yet.

File: util.go [hunk 1]
```diff
@@ -7,21 +7,20 @@ import (
...
-	"log"
 	"os"
 	"path/filepath"
 	"strings"
 )
 
-func ChecksumFile(file string) string {
+func ChecksumFile(file string) (string, error) {
 	hasher := sha256.New()
 	s, err := ioutil.ReadFile(file)
```
Comment (igo95862) [inline util.go:17]: Wasn't ioutil supposed to be removed?

Comment (Foxboron) [reply util.go:17]: Yes, but I thought I'd leave them alone since you had sent a pull-request for it :) I haven't merged that working tree towards master.

Comment (igo95862) [reply util.go:17]: Well my pull request was against master. I will have to update it once this gets merged or rebase it against this branch.

Comment (Foxboron) [reply util.go:17]: I'll do the merging work :) No need to worry about it.

File: cmd/sbctl/create-keys.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	Short: "Create a set of secure boot signing keys",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if !sbctl.CheckIfKeysInitialized(sbctl.KeysPath) {
+			logging.Print("Creating secure boot keys...")
+			err := sbctl.InitializeSecureBootKeys(sbctl.DatabasePath)
+			if err != nil {
+				return fmt.Errorf("couldn't initialize secure boot: %w", err)
+			}
+		} else {
+			logging.Ok("Secure boot keys has already been created!")
```
Comment (WhyNotHugo) [inline cmd/sbctl/create-keys.go:22]: Shouldn't this be fatal (and exit non-zero)?

**Pull request merged**